artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -2370,10 +2370,10 @@ private void deliver() {
                }
             }
 
-            if (pos == endPos) {
-               // Round robin'd all
+            if (pos == endPos || groupConsumer != null) {
```

Comment:
groupConsumer is only set if a msg group is already assigned, or has succesfully handled. i would change this to check if groupID is not null.